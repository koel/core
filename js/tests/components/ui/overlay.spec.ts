import each from 'jest-each'
import Component from '@/components/ui/overlay.vue'
import { mount } from '@/tests/adapter'

describe('components/shared/overlay', () => {
  it('shows with default options', async () => {
    const wrapper = mount(Component)
    // @ts-ignore
    wrapper.vm.show()

    await wrapper.vm.$nextTick()
    expect(wrapper).toMatchSnapshot()
  })

  it('allows option overriding', async () => {
    const wrapper = mount(Component)
    // @ts-ignore
    wrapper.vm.show({
      dismissable: true,
      type: 'warning',
      message: 'Foo'
    })

    await wrapper.vm.$nextTick()
    expect(wrapper).toMatchSnapshot()
  })

  each([['show', true], ['hide', false]]).test('%ss', (methodName) => {
    const wrapper = mount(Component)
    // @ts-ignore
    wrapper.vm[methodName]()
    expect(wrapper).toMatchSnapshot()
  })

  it('dismisses', () => {
    const wrapper = mount(Component)
    // @ts-ignore
    wrapper.vm.show({ dismissable: true })
    expect(wrapper.has('.display')).toBe(true)
    wrapper.click('button.btn-dismiss')
    expect(wrapper.has('.display')).toBe(false)
  })
})
