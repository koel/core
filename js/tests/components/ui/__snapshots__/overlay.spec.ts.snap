// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/shared/overlay allows option overriding 1`] = `
<div id="overlay" class="overlay warning">
  <div class="display">
    <!----> <i class="fa fa-exclamation-circle" style="display: none;"></i> <i class="fa fa-exclamation-triangle" style=""></i> <i class="fa fa-info-circle" style="display: none;"></i> <i class="fa fa-check-circle" style="display: none;"></i> <span class="message">Foo</span></div> <button class="btn-dismiss">Close</button>
</div>
`;

exports[`components/shared/overlay hides 1`] = ``;

exports[`components/shared/overlay shows 1`] = `
<div id="overlay" class="overlay loading">
  <div class="display">
    <div class="bars"><img src="@/../img/bars.gif" alt="Sound bars" height="13" width="auto"></div> <i class="fa fa-exclamation-circle" style="display: none;"></i> <i class="fa fa-exclamation-triangle" style="display: none;"></i> <i class="fa fa-info-circle" style="display: none;"></i> <i class="fa fa-check-circle" style="display: none;"></i> <span class="message"></span>
  </div>
  <!---->
</div>
`;

exports[`components/shared/overlay shows with default options 1`] = `
<div id="overlay" class="overlay loading">
  <div class="display">
    <div class="bars"><img src="@/../img/bars.gif" alt="Sound bars" height="13" width="auto"></div> <i class="fa fa-exclamation-circle" style="display: none;"></i> <i class="fa fa-exclamation-triangle" style="display: none;"></i> <i class="fa fa-info-circle" style="display: none;"></i> <i class="fa fa-check-circle" style="display: none;"></i> <span class="message"></span>
  </div>
  <!---->
</div>
`;
